sdk/log/crate/src/logger.rs

```diff
@@ -560,6 +595,8 @@ macro_rules! impl_log_for_slice {
                     }
                 }
 
+                // JC: shouldn't this call `write_with_args` instead? Or should
+                // this function be called `debug`? Or both?
```

Comment:
Yeah, this is a bit unclear: when you are logging a slice of `str`, the `str` values are logged in "debug" mode, i.e., the values are between quotes (`"`). This is to mimic the behaviour of `format!` for slices, where arrays are formatted with the debug flag `{:?}`.